ODIN_II/parse_odin_stats.py

```diff
@@ -0,0 +1,948 @@
+                return_str += stringify(value_map[index], key) + delim
+        else:
+            for values in value_map:
+                if key in values:
+                    return_str += "NaN" + delim
+                    break
+    # lose the last comma
+    return return_str
+
+# only include keys that have at least one valid entry (but only first one for some reason)
+def get_all_valid_key(value_map, key_list):
+    return_str = ""
+    for key in key_list:
+        if(len(value_map) > 1): # I can't remember what this is for
+            if key in value_map[0] or key in value_map[1]:
+                return_str += key + delim
+        else:
+            return_str += (key + delim) if (key in value_map[0]) else "" # I don't know why this does this
+    return return_str
+
+# TODO this should replace the above function I just don't have time to break the above function, also make this more pythonic
+def get_all_valid_key_multi(value_map, key_list):
+    return_str = ""
+    metric_absent_all = True
+    for key in key_list:
+        i = 0
+        while(i < len(value_map)):
+            if key in value_map[i]:
+                metric_absent_all = False
+            i += 1
+        if(not metric_absent_all):
+            return_str += (key + delim)
+        metric_absent_all = True
+    return return_str
+
+# old function, remove or update later
+def get_all_key(value_map, key_list):
+    return_str = ""
+    for key in key_list:
+        return_str += key + "\t"
+
+    return return_str
+
+# experimental: something that search directory for threshold file, so this can work in a testing suite
+# for now not going to deal with ratios or anything just raw +/- values
+# TODO make thresh name not hardcoded, change name of function
+def threshold_test():
+    thresholds = {}
+    try:
+        file = threshold_file if(threshold_file != "") else "thresholds.txt"
+        ft = open(file, "r")
+        if ft == None:
+            print("thresholds.txt file not included")
+            return
+        else:
+            for wholeLine in ft:
+                parse_line(thresholds, wholeLine)
+    except:
+        print("unable to open threshold file", file)
+        return
+    return thresholds
+
+#   slightly redundant but will be useful in producing our final non-redundant output
+#   checks if value was flagged as +/- outside of threshold or changed to or from 0
+def is_flagged(inLimit, sym1, sym2):
+    if(inLimit == "No"):
+        return True
+    elif((sym1 and not sym2) or (sym2 and not sym1)):
+        return True
+    else:
+        return False
+
+# many different ways to approach this, unclear which is right way, just proof of concept, for now it flags anything +/- threshold limit
+# Change to not be abs value if only increases should be flagged
+def within_threshold(thresholds, category, diff, percentChange):
+    ret_str = ""
+    limit = thresholds.get(category, "N/A") #this doesn't work as hoped
+    if limit == "N/A":
```

Comment:
use None rather than "" and "N/A" to keep it more consistent
i.e
ret_str = None
limit == None